File: evals/refs/sig.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

func main() {
	in := bufio.NewScanner(os.Stdin)
	in.Buffer(make([]byte, 1<<20), 1<<20)
	in.Split(bufio.ScanWords)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	read := func() string { in.Scan(); return in.Text() }

	T, _ := strconv.Atoi(read())
	for c := 0; c < T; c++ {
		N, _ := strconv.Atoi(read())
		parts := make([]string, N)
		for i := 0; i < N; i++ {
			x, _ := strconv.ParseFloat(read(), 64)
			var s float64
			if x >= 0 {
				s = 1.0 / (1.0 + math.Exp(-x))
			} else {
				ex := math.Exp(x)
				s = ex / (1.0 + ex)
			}
			parts[i] = fmt.Sprintf("%.9f", s)
		}
		fmt.Fprintln(out, strings.Join(parts, " "))
	}
}

File: evals/refs/sig.py
import sys, math
data = sys.stdin.read().split()
i = 0
T = int(data[i]); i += 1
out_lines = []
for _ in range(T):
    N = int(data[i]); i += 1
    xs = [float(data[i+j]) for j in range(N)]; i += N
    parts = []
    for x in xs:
        if x >= 0:
            s = 1.0 / (1.0 + math.exp(-x))
        else:
            ex = math.exp(x)
            s = ex / (1.0 + ex)
        parts.append(f'{s:.9f}')
    out_lines.append(' '.join(parts))
sys.stdout.write('\n'.join(out_lines) + '\n')
